src/scripts/main.js

```diff
@@ -1,3 +1,25 @@
 'use strict';
 
 // write code here
+const ulLists = document.querySelectorAll('li');
+
+ulLists.forEach((el) => {
+  if (el.querySelector('ul')) {
+    const textNode = el.firstChild;
```

Comment:
The `firstChild` property may not always return a text node, especially if there are other elements or whitespace before the text. Consider using `firstChild.nodeType === Node.TEXT_NODE` to ensure you are working with a text node.